### PR TITLE
Update UCX shuffle doc w/ memlock limit config [skip ci]

### PR DESCRIPTION
**Pull request by pxLi**
Signed-off-by: Peixin Li <pxli@nyu.edu>

found in our regular doc verification CI. In RDMA test, the client side could fail intermittently,
```
ERROR mlx5_0: ibv_create_cq() failed: Cannot allocate memory.
```

set memlock to unlimited for container should resolve this issue

### CODE REVIEW

**Comment by pxLi** [comment]
build

File: docs/additional-functionality/rapids-shuffle.md [hunk 1]
```diff
@@ -172,7 +172,8 @@ want to take advantage of PCIe peer-to-peer or NVLink need to be visible within
 example, if two containers are trying to communicate and each have an isolated GPU, the link between
 these GPUs will not be optimal, forcing UCX to stage buffers to the host or use TCP.
 Additionally, if you want to use RoCE/Infiniband, the `/dev/infiniband` device should be exposed
-in the container.
+in the container. Also, to avoid potential `failed: Cannot allocate memory`,
+please specify `--ulimit memlock=-1` to make max locked memory as `unlimited` in container.
```
Comment (abellina) [inline docs/additional-functionality/rapids-shuffle.md:176]:
```suggestion
please consider raising the `memlock` ulimit in the container via `--ulimit memlock=[maximum]`. Note that setting `--ulimit memlock=-1` disables the limit.
```

**Review by abellina — APPROVED** [review]
Nice catch. So by default in my box, the container can lock 64MB:
```
$ sudo docker run  busybox sh -c  "ulimit -a"|grep max\ locked
max locked memory (kb)          (-l) 65536
```

And I double checked that your suggestion marks it unlimited:
```
$ sudo docker run --ulimit memlock=-1 busybox sh -c  "ulimit -a"|grep max\ locked
max locked memory (kb)          (-l) unlimited
```

**Comment by pxLi** [comment]
build

**Pull request merged**